src/openfermion/resource_estimates/pbc/thc/factorizations/isdf_test.py

```diff
@@ -0,0 +1,672 @@
+                                      grid_points=grid_points)
+    assert Theta.shape == (len(grid_points), num_interp_points)
+    # Check overlap
+    # Evaluate overlap from orbitals. Do it integral way to ensure
+    # discretization error is the same from using coarse FFT grid for testing
+    # speed
+    ovlp_ao = np.einsum(
+        "mp,mq,m->pq",
+        orbitals.conj(),
+        orbitals,
+        grid_inst.weights,
+        optimize=True,
+    )
+    ovlp_mo = np.einsum("pi,pq,qj->ij",
+                        mf.mo_coeff.conj(),
+                        ovlp_ao,
+                        mf.mo_coeff,
+                        optimize=True)
+    ovlp_mu = np.einsum("Rm,R->m", Theta, grid_inst.weights, optimize=True)
+    orbitals_mo_interp = orbitals_mo[interp_indx]
+    ovlp_isdf = np.einsum(
+        "mi,mj,m->ij",
+        orbitals_mo_interp.conj(),
+        orbitals_mo_interp,
+        ovlp_mu,
+        optimize=True,
+    )
+    assert np.allclose(ovlp_mo, ovlp_isdf)
+    # Check ERIs.
+    eri_ref = mf.with_df.ao2mo(mf.mo_coeff)
+    from pyscf import ao2mo
+
+    eri_ref = ao2mo.restore(1, eri_ref, num_mo)
+    # THC eris
+    Lijn = np.einsum("mi,mj,mn->ijn", chi.conj(), chi, zeta, optimize=True)
+    eri_thc = np.einsum("ijn,nk,nl->ijkl", Lijn, chi.conj(), chi, optimize=True)
+    assert np.allclose(eri_thc, eri_ref)
+
+
+@pytest.mark.skipif(not HAVE_DEPS_FOR_RESOURCE_ESTIMATES,
+                    reason='pyscf and/or jax not installed.')
+def test_supercell_isdf_complex():
+    cell = gto.Cell()
+    cell.atom = """
+    C 0.000000000000   0.000000000000   0.000000000000
+    C 1.685068664391   1.685068664391   1.685068664391
+    """
+    cell.basis = "gth-szv"
+    cell.pseudo = "gth-hf-rev"
+    cell.a = """
+    0.000000000, 3.370137329, 3.370137329
+    3.370137329, 0.000000000, 3.370137329
+    3.370137329, 3.370137329, 0.000000000"""
+    cell.unit = "B"
+    cell.mesh = [11] * 3
+    cell.verbose = 0
+    cell.build(parse_arg=False)
+    # kpts = cell.make_kpts(kmesh, scaled_center=[0.2, 0.3, 0.5])
+
+    mf = scf.RHF(cell, kpt=np.array([0.1, -0.001, 0.022]))
+    mf.kernel()
+    assert np.max(np.abs(mf.mo_coeff.imag)) > 1e-2
+
+    from pyscf.pbc.dft import gen_grid
+
+    grid_inst = gen_grid.UniformGrids(cell)
+    grid_points = cell.gen_uniform_grids(mf.with_df.mesh)
+    orbitals = numint.eval_ao(cell, grid_points, kpt=mf.kpt)
+    orbitals_mo = np.einsum("Rp,pi->Ri", orbitals, mf.mo_coeff, optimize=True)
+    num_mo = mf.mo_coeff.shape[1]
+    num_interp_points = 10 * num_mo
+    nocc = cell.nelec[0]
+    density = np.einsum(
+        "Ri,Ri->R",
+        orbitals_mo[:, :nocc].conj(),
+        orbitals_mo[:, :nocc],
+        optimize=True,
+    )
+    kmeans = KMeansCVT(grid_points, max_iteration=500)
+    interp_indx = kmeans.find_interpolating_points(
+        num_interp_points,
+        density.real,
+        verbose=False,
+    )
+
+    chi, zeta, Theta = supercell_isdf(mf.with_df,
```

Comment:
why is theta capitalized here?

Reply:
ugh, thought I got them all. I'll give it another pass.

Reply:
I'll do the rename in 2,3,4